cita-executor/core/src/libexecutor/amend.rs

```diff
@@ -0,0 +1,170 @@
+use std::cell::RefCell;
+use std::fmt;
+use std::sync::Arc;
+
+use cita_types::{H160, H256, U256};
+use cita_vm::state::{State, StateObjectInfo};
+use util::Bytes;
+
+///amend the abi data
+const AMEND_ABI: u32 = 1;
+///amend the account code
+const AMEND_CODE: u32 = 2;
+///amend the kv of db
+const AMEND_KV_H256: u32 = 3;
+///amend get the value of db
+const AMEND_GET_KV_H256: u32 = 4;
+///amend account's balance
+const AMEND_ACCOUNT_BALANCE: u32 = 5;
+
+pub struct Amend<B> {
+    state_provider: Arc<RefCell<State<B>>>,
+}
+
+impl<B: cita_trie::DB + 'static> Amend<B> {
+    pub fn new(state: Arc<RefCell<State<B>>>) -> Self {
+        Self {
+            state_provider: state,
+        }
+    }
+
+    pub fn call_amend_data(
+        &mut self,
+        value: U256,
+        data: Option<Bytes>,
+    ) -> Result<AmendResult, AmendError> {
+        let amend_type = value.low_u32();
+        match amend_type {
+            AMEND_ABI => self.transact_set_abi(&(data.to_owned().unwrap())),
```

Comment:
maybe it's better not to use `transact_set/get`. It's nothing about `amend`. Why not use the same function with `amend_type`?